File: day-1/dennis.cpp
#include <fstream>
#include <iostream>

using namespace std;

int fuel_tot(const int& fuel) {
    int curr_fuel = (fuel/3)-2;
    if (curr_fuel > 0) {
        return curr_fuel + fuel_tot(curr_fuel);
    }
    return 0;
}

int main() {
    ifstream infile("data/input.txt");
    int fuel;
    int fuel_sum1 = 0;
    int fuel_sum2 = 0;
    while (infile >> fuel) {
        fuel_sum1 += (fuel/3)-2;
        fuel_sum2 += fuel_tot(fuel);
    }

    cout << "D1P1: " << fuel_sum1 << endl;
    cout << "D1P2: " << fuel_sum2 << endl;
    return 0;
}

File: day-1/dennis.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sun Dec  1 10:51:51 2019

@author: dennis
"""

with open('data/input.txt', 'r') as f:
    data = [int(line) for line in f.readlines()]

print("Part 1: {}".format(sum([int(fuel/3)-2 for fuel in data])))

def tot_fuel(fuel):
    curr_fuel = int(fuel/3)-2
    if curr_fuel > 0:
        return curr_fuel + tot_fuel(curr_fuel)
    return 0

print("Part 2: {}".format(sum([tot_fuel(fuel) for fuel in data])))
